app/src/main/kotlin/com/intive/picover/profile/viewmodel/ProfileViewModel.kt

```diff
@@ -25,15 +26,10 @@ class ProfileViewModel @Inject constructor(
 	@Validator.ShortText private val textValidator: TextValidator,
 ) : StatefulViewModel<Profile>() {
 
-	val userName = derivedStateOf {
-		_state.value.let { state ->
-			if (state is Loaded) {
-				state.data.name
-			} else {
-				""
-			}
-		}
-	}
+	private val _username = mutableStateOf("")
+	val username: State<String> = _username
+	val usernameErrorMessageId
+		get() = textValidator.validate(username.value).errorMessageId
```

Comment:
if we would use `MutableStateFlow` in viewmodels then we could get rid of such hacks, and use functions like `map`, `combine` which are originally designed for such use cases. I don't block this pr, topic to discuss